### PR TITLE
Add top level system test

### PR DESCRIPTION
**Pull request by mkhansen-intel**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | N/A ||
| Primary OS tested on | Ubuntu 18.04 |
| Robotic platform tested on | Runs using Gazebo |

---

## Description of contribution in a few bullet points
* Adds system test which launches Gazebo server with turtlebot3 world plus entire stack
* Sets use_sim_time for transforms and map server
* Sends an initial pose and waits for a pose returned from /amcl_pose
* Sends a goal pose and monitors for 30seconds
* If goal is reached within 0.5 meters, test PASSES
* If goal isn't reached it will retry one time

---

## Future work that may be required in bullet points
* ~~Currently still WIP~~ due to instability of the bring up, sometimes it passes, others it doesn't. 
* ~~Not totally tested within colcon test yet~~
* ~~Will resolve those issues before turning on in colcon~~
* ~~Pushed here for other tests (ie. AMCL) to use as a reference / starting point~~
* Change travis.yaml to run the test automatically (future PR)
* Add a bt_navigator test (future PR)

### CODE REVIEW

**Comment by mkhansen-intel** [comment]
All, I think this is ready to review.

File: nav2_system_tests/src/system/test_system_node.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        d_y = self.current_pose.position.y - self.goal_pose.position.y
+        distance = math.sqrt(d_x * d_x + d_y * d_y)
+        self.get_logger().info('Distance from goal is: ' + str(distance))
+        return distance
+
+    def setSimTime(self):
+        self.get_logger().info('Setting transforms to use sim time from gazebo')
+        from subprocess import call
+        # loop through the problematic nodes
+        for nav2_node in ('/static_transform_publisher', '/map_server'):
```
Comment (crdelsey) [inline nav2_system_tests/src/system/test_system_node.py]:
```suggestion
        for nav2_node in ('/static_transform_publisher', '/map_server', '/global_costmap/global_costmap', '/local_costmap/local_costmap'):
```

File: nav2_system_tests/src/system/test_system_launch.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            package='nav2_amcl',
+            node_executable='amcl',
+            node_name='amcl',
+            output='screen',
+            parameters=[{'use_sim_time': use_sim_time}]),
+
+        launch_ros.actions.Node(
+            package='nav2_world_model',
+            node_executable='world_model',
+            node_name='world_model',
```
Comment (crdelsey) [inline nav2_system_tests/src/system/test_system_launch.py]: For world model and dwb controller, you'll need to set parameters using a yaml file now, and you'll want to remove the `node_name` line

Comment (mkhansen-intel) [reply nav2_system_tests/src/system/test_system_launch.py]: Fixed

**Comment by mkhansen-intel** [comment]
I made the required changes. Assuming it passes CI I'd like to merge it.

**Comment by mkhansen-intel** [comment]
The build on 'master' is broken due to #422, @mhpanah is working on a fix. Until then, CI won't pass.

**Comment by mkhansen-intel** [comment]
Passed CI, 2 checkmarks, merging now...

**Pull request merged**